include/aws/crt/mqtt/Mqtt5Client.h

```diff
@@ -435,6 +435,8 @@ namespace Aws
 
                 virtual ~Mqtt5Client();
 
+                struct aws_mqtt5_client *GetUnderlyingHandle() const noexcept;
```

Comment:
My blunt response to that would be something like this: "that scenario represents such a perversely foolish/self-destructive behavior by the user that I have no interest in trying to prevent it."

More generally, getters for the underlying C object is a pattern we use throughout the CRT.